src/utils/Client.util.ts

```diff
@@ -63,6 +67,56 @@ export function createHTTPMetadataQueryParam(metadata: KeyValueType = {}): strin
   return queryParam;
 }
 
+/**
+ * Converts a set of optional parameters into a query string
+ * @param params
+ */
+export function createHTTPStateBehavioralQueryParam(params: Partial<IStateOptions> = {}): string {
+  const optParamsBuilder = new URLSearchParams();
+
+  if (params?.consistency) {
+    const consistency = matchStateConsistency(params?.consistency);
+    if (consistency !== undefined) optParamsBuilder.set("consistency", consistency);
+  }
+
+  if (params?.concurrency) {
+    const concurrency = matchStateConcurrency(params?.concurrency);
+    if (concurrency !== undefined) optParamsBuilder.set("concurrency", concurrency);
+  }
+
+  return optParamsBuilder.toString();
+}
+
+/**
+ * Return the string representation of a valid consistency configuration
+ * @param c
+ */
+export function matchStateConsistency(c: EStateConsistency): "eventual" | "strong" | undefined {
```

Comment:
let's use `getEstateConstencyValue`, but just wondering, couldn't we get the values of an enum directly?

Reply:
I actually tried to simplify this, but here's my reasoning :
Both enums are number enums. These numbers are used as is by the GRPC implementation, but the HTTP implementation requires specific string values. 
Each enum has an "*_unspecified" that can't be used in the HTTP implementation. 
And most importantly, the enum value comes from the user, and since we're using TypeScript <5.0, there's nothing stopping them from using a random integer as a value. 
That's why I've added an extra switch statement in these cases. 
Maybe we could do this differently?